### ISSUE TITLE
Nearby voxels are rendered incorrectly near bottom of the screen

### ISSUE THREAD

**Issue by MrAlaux** [fabiangreffrath/woof#1277]
You may know about this already, but I just wanted to make sure of it.

See for yourself:
![ezgif com-optimize](https://github.com/fabiangreffrath/woof/assets/73968015/9a0ad21c-d53d-4c00-982a-dfcec0c6d098)

**Comment by rfomin** [comment]
The original implementation has the same problem (voxel-rend build): ![image](https://github.com/fabiangreffrath/woof/assets/5077629/9783a374-ef6d-407c-9544-797c461667d1)
It seems that the vertical clipping in `VX_DrawColumn()` is not accurate.

**Comment by NightFright2k19** [comment]
Wait a minute... this thing can render voxels and I don't know about it? <.<

**Comment by fabiangreffrath** [comment]
This will be in the next major release.

**Comment by andwj** [comment]
This happens because the voxel column (on the XY plane) is hitting the viewplane -- at least one coord is behind it, and one in front of it.  The code cannot handle that and it skips drawing that column.

The fix is non-trivial, but I'm thinking about it :smiley:

**Comment by rfomin** [comment]
> The fix is non-trivial, but I'm thinking about it

Thank you! We also found an integer overflow witch happens at high resolutions:
https://github.com/fabiangreffrath/woof/blob/3584cb67bb2298ed39fbc4b5b40ad0bdabed6c17/src/r_voxel.c#L820
Not sure if the fix is correct.

**Comment by MrAlaux** [comment]
Something else about voxels, I'm not sure if it deserves an issue of its own (_tested in the [latest master artifact (Win-64)](https://github.com/fabiangreffrath/woof/actions/runs/10180425769))_:

Fuzzy voxels don't get anything drawn on their under and upper sides:

<details><summary>Screenshots</summary>
<p>

![woof0003](https://github.com/user-attachments/assets/2da491c0-5d1f-483b-a257-d4991a8764a1)
![woof0004](https://github.com/user-attachments/assets/082c7a8c-7bd1-42a1-a01d-a5829a45c2d6)
Thanks to ceski for the test map, by the way.
</p>
</details>

**Comment by fabiangreffrath** [comment]
> Fuzzy voxels don't get anything drawn on their under and upper sides:

Indeed, drawing fuzzy voxels is interrupted before voxel top and bottom are drawn:

https://github.com/fabiangreffrath/woof/blob/6eb4413fb831faf978b1ffb01b3f7e46b1f08875/src/r_voxel.c#L869-L883
